pyk/src/pyk/ktool/kprint.py

```diff
@@ -126,11 +126,13 @@ def build_symbol_table(definition: KDefinition, opinionated=False) -> SymbolTabl
     for module in definition.modules:
         for prod in module.syntax_productions:
             assert(prod.klabel)
+
             label = prod.klabel.name
-            if 'symbol' in prod.att and 'klabel' in prod.att:
-                label = prod.att['klabel']
             unparser = unparser_for_production(prod)
+
             symbol_table[label] = unparser
+            if 'symbol' in prod.att and 'klabel' in prod.att:
+                symbol_table[prod.att['klabel']] = unparser
```

Comment:
This was needed because of some weirdness in unparsing List units:

The production had `prod.klabel.name = List{"_,_"}_Id` and  klabel attribute as `List{"_,_"}`.
The AST nodes had `KApply.label = 'List{"_,_"}'`.

I guess we need to locate the java code for unparsing and duplicate it's logic here.

Reply:
Hmmm, or we need to identify the bug which is giving the wrong labels to productions and ast nodes.

Reply:
Recorded: https://github.com/runtimeverification/k/issues/2801